File: Tarea7/CuentaBanco.java
import java.util.Scanner;

import java.util.Random;
public class CuentaBanco {
    private long numeroCuenta;
    private char tipoCuenta;
    private double saldoCuenta;

    public CuentaBanco() {

    }
    public CuentaBanco(long numeroCuenta, char tipoCuenta) {
        this.numeroCuenta = numeroCuenta;
        this.tipoCuenta = tipoCuenta;
    }
    public void deposito() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Ingresa la cantidad que deseas depositar: ");
        double dinero = scanner.nextDouble();

        if (dinero > 0) {
            if (this.tipoCuenta == 'A' && this.saldoCuenta + dinero <= 50000) {
                this.saldoCuenta += dinero;
                System.out.println("Se han depositado: $" + dinero + " a la cuenta.");
            } else if (this.tipoCuenta == 'B' && this.saldoCuenta + dinero <= 100000) {
                this.saldoCuenta += dinero;
                System.out.println("Se han depositado: $" + dinero + " a la cuenta.");
            } else if (this.tipoCuenta == 'C') {
                this.saldoCuenta += dinero;
                System.out.println("Se han depositado: $" + dinero + " a la cuenta.");
            }
        }
    }
    public void retiro () {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Ingresa la cantidad que deseas retirar: ");
        double dinero = scanner.nextDouble();

        if (tipoCuenta == 'A' && this.saldoCuenta - dinero >= 1000) {
            this.saldoCuenta -= dinero;
            System.out.println("Se retiraron $" + dinero + " de su cuenta");
            System.out.println("Saldo total despues del retiro: $" + this.saldoCuenta);
        } else if (tipoCuenta == 'B' && this.saldoCuenta - dinero >= 5000) {
            this.saldoCuenta -= dinero;
            System.out.println("Se retiraron $" + dinero + " de su cuenta");
            System.out.println("Saldo total despues del retiro: $" + this.saldoCuenta);
        } else if (tipoCuenta == 'C' && this.saldoCuenta - dinero >= 10000) {
            this.saldoCuenta -= dinero;
            System.out.println("Se retiraron $" + dinero + " de su cuenta");
            System.out.println("Saldo total despues del retiro: $" + this.saldoCuenta);
        } else {
            System.out.println("No se puede retirar tal cantidad, su cuenta quedaria con saldo menor al que tiene permitido");
        }
    }
    public long getNumeroCuenta() {
        return numeroCuenta;
    }
    public void setNumeroCuenta() {
        Random random = new Random();
        this.numeroCuenta = random.nextInt(9000) + 1000;
    }
    public char getTipoCuenta() {
        return tipoCuenta;
    }
    public void setTipoCuenta() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Ingresa el tipo de cuenta que desea tener: ");
        this.tipoCuenta =  scanner.next().charAt(0);;
    }
    public double getSaldoCuenta() {
        System.out.println("Este es su saldo: $" + saldoCuenta);
        return saldoCuenta;
    }
    public void setSaldoCuenta() {
        this.saldoCuenta = saldoCuenta + 1000;
    }
}

File: Tarea7Python/CuentaBanco.py
import random

class CuentaBanco:
    def __init__(self, numero_cuenta=0, tipo_cuenta='', saldo_cuenta=0.0):
        self.numero_cuenta = numero_cuenta
        self.tipo_cuenta = tipo_cuenta
        self.saldo_cuenta = saldo_cuenta

    def deposito(self):
        dinero = float(input("Ingresa la cantidad que deseas depositar: "))
        if dinero > 0:
            if self.tipo_cuenta == 'A' and self.saldo_cuenta + dinero <= 50000:
                self.saldo_cuenta += dinero
                print("Se han depositado: $" + str(dinero) + " a la cuenta.")
            elif self.tipo_cuenta == 'B' and self.saldo_cuenta + dinero <= 100000:
                self.saldo_cuenta += dinero
                print("Se han depositado: $" + str(dinero) + " a la cuenta.")
            elif self.tipo_cuenta == 'C':
                self.saldo_cuenta += dinero
                print("Se han depositado: $" + str(dinero) + " a la cuenta.")

    def retiro(self):
        dinero = float(input("Ingresa la cantidad que deseas retirar: "))
        if self.tipo_cuenta == 'A' and self.saldo_cuenta - dinero >= 1000:
            self.saldo_cuenta -= dinero
            print("Se retiraron $" + str(dinero) + " de su cuenta")
            print("Saldo total después del retiro: $" + str(self.saldo_cuenta))
        elif self.tipo_cuenta == 'B' and self.saldo_cuenta - dinero >= 5000:
            self.saldo_cuenta -= dinero
            print("Se retiraron $" + str(dinero) + " de su cuenta")
            print("Saldo total después del retiro: $" + str(self.saldo_cuenta))
        elif self.tipo_cuenta == 'C' and self.saldo_cuenta - dinero >= 10000:
            self.saldo_cuenta -= dinero
            print("Se retiraron $" + str(dinero) + " de su cuenta")
            print("Saldo total después del retiro: $" + str(self.saldo_cuenta))
        else:
            print("No se puede retirar tal cantidad, su cuenta quedaría con saldo menor al que tiene permitido")

    def get_numero_cuenta(self):
        return self.numero_cuenta

    def set_numero_cuenta(self):
        self.numero_cuenta = random.randint(1000, 9999)

    def get_tipo_cuenta(self):
        return self.tipo_cuenta

    def set_tipo_cuenta(self):
        self.tipo_cuenta = input("Ingresa el tipo de cuenta que desea tener: ")

    def get_saldo_cuenta(self):
        print("Este es su saldo: $" + str(self.saldo_cuenta))
        return self.saldo_cuenta

    def set_saldo_cuenta(self):
        self.saldo_cuenta += 1000
